NetTopologySuite.IO.PostGis/PostGisReader.cs

```diff
@@ -211,30 +211,41 @@ protected ICoordinateSequence ReadCoordinateSequenceRing(BinaryReader reader, IC
         /// <param name="factory">The geometry factory to use for geometry creation.</param>
         /// <param name="precisionModel">The precision model used to make x- and y-ordinates precise.</param>
         /// <param name="numPoints">The number of points in the coordinate sequence.</param>
-        /// <param name="ordinates">The ordinates to read. <see cref="Ordinates.XY"/> are always read.</param>
+        /// <param name="receivedOrdinates">The ordinates to read. <see cref="Ordinates.XY"/> are always read.</param>
         /// <returns>The coordinate sequence</returns>
-        protected ICoordinateSequence ReadCoordinateSequence(BinaryReader reader, ICoordinateSequenceFactory factory, IPrecisionModel precisionModel, int numPoints, Ordinates ordinates)
+        protected ICoordinateSequence ReadCoordinateSequence(BinaryReader reader, ICoordinateSequenceFactory factory, IPrecisionModel precisionModel, int numPoints, Ordinates receivedOrdinates)
         {
-            var sequence = factory.Create(numPoints, HandleOrdinates);
+            var outputOrdinates = receivedOrdinates;
+            if (HandleOrdinates != Ordinates.None)
+                outputOrdinates &= HandleOrdinates;
 
-            double ordinateZ = Coordinate.NullOrdinate;
-            double ordinateM = Coordinate.NullOrdinate;
+            var sequence = factory.Create(numPoints, outputOrdinates);
 
-            bool getZ = (ordinates & Ordinates.Z) == Ordinates.Z;
-            bool getM = (ordinates & Ordinates.M) == Ordinates.M;
-
-            bool handleZ = (HandleOrdinates & Ordinates.Z) == Ordinates.Z;
-            bool handleM = (HandleOrdinates & Ordinates.M) == Ordinates.M;
+            var receivedZ = receivedOrdinates.HasFlag(Ordinates.Z);
+            var receivedM = receivedOrdinates.HasFlag(Ordinates.M);
+            var outputtingZ = outputOrdinates.HasFlag(Ordinates.Z);
+            var outputtingM = outputOrdinates.HasFlag(Ordinates.M);
 
             for (int i = 0; i < numPoints; i++)
             {
                 sequence.SetOrdinate(i, Ordinate.X, precisionModel.MakePrecise(reader.ReadDouble()));
                 sequence.SetOrdinate(i, Ordinate.Y, precisionModel.MakePrecise(reader.ReadDouble()));
-                if (getZ) ordinateZ = reader.ReadDouble();
-                if (handleZ) sequence.SetOrdinate(i, Ordinate.Z, ordinateZ);
-                if (getM) ordinateM = reader.ReadDouble();
-                if (handleM) sequence.SetOrdinate(i, Ordinate.M, ordinateM);
+
+                if (receivedZ)
+                {
+                    var z = reader.ReadDouble();
+                    if (outputtingZ)
+                        sequence.SetOrdinate(i, Ordinate.Z, z);
```

Comment:
The precision model must be used here and below.
```suggestion
                        sequence.SetOrdinate(i, Ordinate.Z, precisionModel.MakePrecise(z));
```

Reply:
No, `NetTopologySuite.IO.WKBReader` and `NetTopologySuite.IO.WKTReader` only apply precisison model to x- and y-ordinates:
https://github.com/NetTopologySuite/NetTopologySuite/blob/583ae18b593e5835630aa7e67bb82dd8eb349ccd/NetTopologySuite/IO/WKBReader.cs#L272-L311

This is conformant with JTS.

Reply:
Is compatibility so important? In such cases it sounds like a bug which should be fixed in the two projects.

Reply:
> it sounds like a bug

I don't consider this to be a bug.  AFAIK, JTS / NTS only "really" use the X and Y ordinates for its own operations and Z/M are just sorta... "there".

It's possible that perhaps (often? [usually?]) Z/M are at a different scale than X/Y, and so the same `PrecisionModel` might not even be appropriate for them.

Reply:
I tend to agree with @YohDeadfall that it's strange and quite error-prone to treat Z/M differently in any way from X/Y, regardless of whether JTS/NTS perform operations on them or not... Even if Z/M may have different scales, it doesn't seem like that's necessarily the case, and support is needed for different precision models for different ordinates, then that's a new feature...

However, you guys are the GIS experts and in any case this is out-of-scope from the changes of this PR... We can continue discussion in a new issue.

Reply:
> We can continue discussion in a new issue

I've opened NetTopologySuite/NetTopologySuite#280 for this.